commands/generate.go

```diff
@@ -297,3 +302,67 @@ func generateknativev1alpha1ServingCRDYAML(services stack.Services, format schem
 
 	return objectsString, nil
 }
+
+func generateRioCRDYAML(services stack.Services, format schema.BuildFormat, apiVersion, namespace, branch, version string) (string, error) {
+	crds := []riov1.CRD{}
+
+	for name, function := range services.Functions {
+
+		fileEnvironment, err := readFiles(function.EnvironmentFile)
+		if err != nil {
+			return "", err
+		}
+
+		//combine all environment variables
+		allEnvironment, envErr := compileEnvironment([]string{}, function.Environment, fileEnvironment)
+		if envErr != nil {
+			return "", envErr
+		}
+
+		var env []riov1.EnvPair
+		for k, v := range allEnvironment {
+			env = append(env, riov1.EnvPair{Name: k, Value: v})
+		}
+
+		var ports []riov1.Port
+		ports = append(ports, riov1.Port{Port: 8080, Protocol: "HTTP", TargetPort: 8080})
+
+		crd := riov1.CRD{
+			Metadata: schema.Metadata{
+				Name:      name,
+				Namespace: namespace,
+			},
+			APIVersion: apiVersion,
+			Kind:       "Service",
+			Spec: riov1.Spec{
+				Env: env,
+				Concurrency: 10,
+				Cpus: "0",
+				Image: function.Image,
+				ImagePullPolicy: "IfNotPresent",
+				MaxScale: 1,
+				MinScale: 1,
+				Ports: ports,
+				Rollout: true,
+				RolloutIncrement: 5,
```

Comment:
What do we get if we leave these two Rollout settings blank along with the Weight? Do they default?

If so, let's fill out the absolute bare minimum. That way, if @ibuildthecloud changes the defaults we get them "for free"

Reply:
I'll check this

Reply:
I've removed all except ports, image and env (if specified in stack.yml)